src/controller/python/OpCredsBinding.cpp

```diff
@@ -319,6 +322,68 @@ void pychip_OnCommissioningStatusUpdate(chip::PeerId peerId, chip::Controller::C
+                                                                        uint32_t serializedEphemeralKeyLen, uint8_t * noc,
+                                                                        uint32_t nocLen, uint8_t * icac, uint32_t icacLen,
+                                                                        uint8_t * rcac, uint32_t rcacLen,
+                                                                        chip::VendorId adminVendorId, bool enableServerInteractions)
+{
+    ChipLogDetail(Controller, "Creating New Device Controller");
+
+    auto devCtrl = std::make_unique<chip::Controller::DeviceCommissioner>();
+    VerifyOrReturnError(devCtrl != nullptr, ToPyChipError(CHIP_ERROR_NO_MEMORY));
+
+    chip::Crypto::P256Keypair ephemeralKey;
+    chip::Crypto::P256SerializedKeypair keyPair;
+    memcpy(keyPair.Bytes(), serializedEphemeralKey, serializedEphemeralKeyLen);
+    CHIP_ERROR err = ephemeralKey.Deserialize(keyPair);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, ToPyChipError(err));
+
+    ReturnErrorCodeIf(nocLen > Controller::kMaxCHIPDERCertLength, ToPyChipError(CHIP_ERROR_NO_MEMORY));
+    ReturnErrorCodeIf(icacLen > Controller::kMaxCHIPDERCertLength, ToPyChipError(CHIP_ERROR_NO_MEMORY));
+    ReturnErrorCodeIf(rcacLen > Controller::kMaxCHIPDERCertLength, ToPyChipError(CHIP_ERROR_NO_MEMORY));
+
+    Controller::SetupParams initParams;
+    initParams.pairingDelegate                = &sPairingDelegate;
+    initParams.operationalCredentialsDelegate = &sDummyOperationalCredentialsIssuer;
+    initParams.operationalKeypair             = &ephemeralKey;
+    initParams.controllerRCAC                 = ByteSpan(rcac, rcacLen);
+    initParams.controllerICAC                 = ByteSpan(icac, icacLen);
+    initParams.controllerNOC                  = ByteSpan(noc, nocLen);
+    initParams.enableServerInteractions       = enableServerInteractions;
+    initParams.controllerVendorId             = adminVendorId;
+    initParams.permitMultiControllerFabrics   = true;
+
+    err = Controller::DeviceControllerFactory::GetInstance().SetupCommissioner(initParams, *devCtrl);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, ToPyChipError(err));
+
+    // Setup IPK in Group Data Provider for controller after Commissioner init which sets-up the fabric table entry
+    uint8_t compressedFabricId[sizeof(uint64_t)] = { 0 };
+    chip::MutableByteSpan compressedFabricIdSpan(compressedFabricId);
+
+    err = devCtrl->GetCompressedFabricIdBytes(compressedFabricIdSpan);
+    VerifyOrReturnError(err == CHIP_NO_ERROR, ToPyChipError(err));
+
+    ChipLogProgress(Support, "Setting up group data for Fabric Index %u with Compressed Fabric ID:",
+                    static_cast<unsigned>(devCtrl->GetFabricIndex()));
+    ChipLogByteSpan(Support, compressedFabricIdSpan);
+
+    chip::ByteSpan defaultIpk = chip::GroupTesting::DefaultIpkValue::GetDefaultIpk();
```

Comment:
Must allow the IPK to be provided as a parameter to the init. This is critical to be able to work with our internal CA

Reply:
Please let the IPK be passed-on

Reply:
Added a param for IPK